sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2360,13 +2360,25 @@ object GpuOverrides {
       " Returns an array with the given elements",
       ExprChecks.projectNotLambda(
         TypeSig.ARRAY.nested(TypeSig.numeric + TypeSig.NULL + TypeSig.STRING +
-            TypeSig.BOOLEAN + TypeSig.DATE + TypeSig.TIMESTAMP),
+            TypeSig.BOOLEAN + TypeSig.DATE + TypeSig.TIMESTAMP + TypeSig.ARRAY),
         TypeSig.ARRAY.nested(TypeSig.all),
         repeatingParamCheck = Some(RepeatingParamCheck("arg",
           TypeSig.numeric + TypeSig.NULL + TypeSig.STRING +
-              TypeSig.BOOLEAN + TypeSig.DATE + TypeSig.TIMESTAMP,
+              TypeSig.BOOLEAN + TypeSig.DATE + TypeSig.TIMESTAMP +
+              TypeSig.ARRAY.nested(TypeSig.numeric + TypeSig.NULL + TypeSig.STRING +
+                TypeSig.BOOLEAN + TypeSig.DATE + TypeSig.TIMESTAMP + TypeSig.ARRAY),
           TypeSig.all))),
       (in, conf, p, r) => new ExprMeta[CreateArray](in, conf, p, r) {
+
+        override def tagExprForGpu(): Unit = {
+          wrapped.dataType match {
+            case ArrayType(ArrayType(ArrayType(_, _), _), _) =>
```

Comment:
Where did this limitation come from? Does cudf only support so much nesting?

Reply:
I tested like array(array(array(a))) and libcudf will throw the exception, I read the PR for making list and found there is a comment  in  `Currently, only lists columns of one depth level are supported.` https://github.com/rapidsai/cudf/pull/8046

Reply:
Could you file a follow on issue for more deeply nested? It is not something we have to work on. I just want to be sure that it is documented.

Reply:
sure, filed the following issue https://github.com/NVIDIA/spark-rapids/issues/2453